tools/codegen/api/autograd.py

```diff
@@ -120,3 +120,94 @@ class DifferentiableOutput:
 
     # TODO: only to keep it byte-for-byte compatible with the old codegen, should remove.
     cpp_type: str
+
+@dataclass(frozen=True)
+class NativeFunctionWithDifferentiabilityInfo:
+    func: NativeFunction
+    info: Optional[DifferentiabilityInfo]
+
+def dispatch_strategy(fn: NativeFunctionWithDifferentiabilityInfo) -> str:
```

Comment:
Since we're moving this, it might be the right time to strip the old comments (they're still from the world of Tensors and Variables).

I'm actually not even sure the right way to describe this function's role in the current system, but it's probably out of scope for this PR to, like, recontexualize it. Maybe just a TODO above the comment so people reading don't try to make sense of it w.r.t. the current system... before this PR they could at least look at the history and see that it's like 3 years old, but afterwards there's nothing to tell them it's not fresh code.